pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -537,17 +551,17 @@ func (p *PriorityQueue) Update(oldPod, newPod *v1.Pod) error {
 
 	if oldPod != nil {
 		// If the pod is already in the active queue, just update it there.
-		if _, exists, _ := p.activeQ.Get(oldPod); exists {
+		if _, exists, _ := p.activeQ.Get(newPodInfoNoTimestamp(oldPod)); exists {
 			p.nominatedPods.update(oldPod, newPod)
-			err := p.activeQ.Update(newPod)
+			err := p.activeQ.Update(p.newPodInfo(newPod))
 			return err
 		}
 
 		// If the pod is in the backoff queue, update it there.
-		if _, exists, _ := p.podBackoffQ.Get(oldPod); exists {
+		if _, exists, _ := p.podBackoffQ.Get(newPodInfoNoTimestamp(oldPod)); exists {
 			p.nominatedPods.update(oldPod, newPod)
-			p.podBackoffQ.Delete(newPod)
-			err := p.activeQ.Add(newPod)
+			p.podBackoffQ.Delete(newPodInfoNoTimestamp(oldPod))
+			err := p.activeQ.Add(p.newPodInfo(newPod))
```

Comment:
We should keep existing timestamp here too.

Reply:
Done